src/components/AppIcon.js

```diff
@@ -2,20 +2,37 @@ import React, { Component } from "react";
 import Ink from "react-ink";
 import styles from "./AppIcon.css";
 import { withRouter } from "react-router-dom";
+import { injectIntl } from "react-intl";
 
 
 class AppIcon extends Component {
   render() {
-    const { src, title, subTitle, handleClick } = this.props;
+    const { src, title, subTitle, handleClick, readonly, requiredRoles } = this.props;
+    const requiresRoleMessage = this.props.intl.formatMessage({ id: "authorization.requires_role", defaultMessage: "Requires role: " });
+    const requiresRolesMessage = this.props.intl.formatMessage({ id: "authorization.requires_roles", defaultMessage: "Requires one of the following roles: " });
```

Comment:
Doesn't react-intl have a way to deal with pluralization, besides multiple messages? Most translation libraries do.

Reply:
changed.
I am always very hesitant to put additional logic in something like a translation library.
It adds plenty of technical complexity.